chapters/chap03-sec02.tex

```diff
@@ -18,17 +18,97 @@ \subsection{\mimc{} Encryption}\label{instantiation:mkhash:mimc-encryption}
+
+\paragraph{Statistical attacks.}
+
+In the ``Security Analysis'' section~\cite[Section 4.2]{albrecht2016mimc}, the authors explain that since the cubic function is an APN, linear attacks pose no threat to $\mimc{}$.
+
+We observe that this claim aligns with~\cite[Theorem 2]{DBLP:journals/tit/HellesethRS99}. In fact if $\mimc{}$ is operated over $\FFx{2^n}$, it is easy to see that the degree of the cubic map can be expressed as $3 = 2^t + 1$, where $t = 1$, and where $n$ and $1$ are trivially coprime. This case is covered by~\cite[Theorem 2]{DBLP:journals/tit/HellesethRS99} which confirms that the cubic function $S(x) = x^3$ over $\FFx{2^n}$ is an APN/2-uniform mapping, as desired for differential and linear cryptanalysis resistance.
+
+Likewise, in~\cite[Section 5.1]{albrecht2016mimc} the authors claim that, provided that the cubic map is a permutation over the prime field of interest, then $\mimc{}$ can be used to operate over prime fields of odd characteristic. In this case too, $S(x) = x^3$ is an APN, provided $p \neq 3$ (as reported by~\cite[Theorem 3, item 3]{DBLP:journals/tit/HellesethRS99}).
+
+Furthermore, while~\cite[Section 5.3]{albrecht2016mimc} proposes to relax the choice of the map degree to be of the general form $2^t \pm 1$, so long as the map remains a permutation, the authors showed that, out of these ``exponent shapes'', the case $\exponent{} = 2^t + 1$ is, unfortunately, not as good as it initially seems, in $\FFx{2^n}$, due to term cancellation in fields of characteristic 2 that renders the resulting polynomial ``sparse''\footnote{Since the round function is a polynomial, the whole scheme can be seen as a polynomial which overall degree and ``sparsity'' depend on the underlying field characteristic, degree of the round function and number of rounds.} (the degree of the polynomial will be bounded by $3^r$, $r$ being the number of rounds, which does not constitute an improvement on the status quo (i.e. $\exponent{} = 3$)). For this very reason, exponents of the form $2^t + 1, t > 1$, may not be of interest (sparse polynomial and more expensive arithmetic in the round function).
+
+Likewise, if the map degree $\exponent{}$ is chosen to be of the form $2^t + 1$, with $\gcd(\exponent{}, 2^n - 1) = 1$ in the context of $\mimc{}$ over $\FFx{2^n}$, it is necessary to bear in mind that, without the extra requirement that $t$ needs to be coprime with $n$, then this case is not covered by~\cite[Theorem 2]{DBLP:journals/tit/HellesethRS99}, and $S(x) = x^\exponent{}$ does not have differential 2-uniformity anymore - violating the claim made in~\cite[Section 4.2]{albrecht2016mimc} paragraph ``Linear Attacks'' about optimal resistance against linear and differential cryptanalysis. (In fact, depending on the value of $g = gcd(n, t)$, the map $S(x) = x^{2^t + 1}$ would be differentially $2^g$-uniform~\cite{DBLP:conf/eurocrypt/Nyberg93} - contrasting with the setting considered in paragraph ``Linear attacks'' where $\exponent{} = 3$).
```

Comment:
> does not have differential 2-uniformity
I find that very confusing as differential 2-uniformity" has not been defined :/

Reply:
https://github.com/clearmatics/zeth-specifications/pull/6#discussion_r607898289